x/auth/ante/validator_tx_fee.go

```diff
@@ -12,36 +13,39 @@ import (
 
 // checkTxFeeWithValidatorMinGasPrices implements the default fee logic, where the minimum price per
 // unit of gas is fixed and set by each validator, can the tx priority is computed from the gas price.
-func checkTxFeeWithValidatorMinGasPrices(ctx sdk.Context, tx sdk.Tx) (sdk.Coins, int64, error) {
-	feeTx, ok := tx.(sdk.FeeTx)
+func checkTxFeeWithValidatorMinGasPrices(ctx sdk.Context, tx sdk.Tx, params types.Params) (sdk.Coins, int64, error) {
+	amount, ok := sdkmath.NewIntFromString(params.GetTxFees())
 	if !ok {
-		return nil, 0, errorsmod.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
+		return nil, 0, errorsmod.Wrap(sdkerrors.ErrInvalidTxFees, "must provide correct txFees")
 	}
 
-	feeCoins := feeTx.GetFee()
-	gas := feeTx.GetGas()
+	// TODO HV2: gas is retrieved from Params as currently done in heimdall. Can this be changed
+	gas := params.GetMaxTxGas()
+	feeCoins := sdk.Coins{sdk.Coin{Denom: types.FeeToken, Amount: amount}}
 
+	// TODO HV2: removed as not present in heimdall. Is this safe? Would it be better/safer to implement it?
```

Comment:
I see a similar hunk was removed in original Heimdall, in file `x/auth/ante.go`, function `NewAnteHandler`. It would be good to compare the code that was removed there with this one to check they have the same functionality.

Reply:
In my understanding, it's the same business logic. Will keep it commented out, and update the comment

Reply:
Resolved